### ISSUE TITLE
Deployed pods missing workloadID label - no active endpoints - 503 error

### ISSUE THREAD

**Issue by rossigee** [rancher/rancher#16052]
**Rancher versions:**
rancher/server or rancher/rancher: 2.1.0
rancher/agent or rancher/rancher-agent: 2.1.0

**Infrastructure Stack versions:**
healthcheck:
ipsec:
network-services:
scheduler:
kubernetes (if applicable): 1.12.0

**Docker version: (`docker version`,`docker info` preferred)**

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

Single node rancher.

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**

Kubernetes

**Steps to Reproduce:**

(originally filed as https://github.com/kubernetes/kubernetes/issues/69563 but I now suspect that the missing label may be related to R2)

We deploy a new version of our app by changing the `spec.template.spec.containers[0].image` attribute of the Deployment YAML, as described in [the documentation](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#updating-a-deployment) for `Deployment` controllers...

The `Deployment` YAML looks like this:

```
apiVersion: apps/v1beta2
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "17"
    field.cattle.io/creatorId: user-5jgmc
  creationTimestamp: 2018-10-02T04:54:33Z
  generation: 45
  labels:
    workload.user.cattle.io/workloadselector: deployment-cms-app
  name: app
  namespace: cms
  resourceVersion: "175227"
  selfLink: /apis/apps/v1beta2/namespaces/cms/deployments/app
  uid: 40eeb300-c5ff-11e8-91dc-001b21dc82ba
spec:
  minReadySeconds: 5
  progressDeadlineSeconds: 60
  replicas: 3
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      workload.user.cattle.io/workloadselector: deployment-cms-app
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 0
    type: RollingUpdate
  template:
    metadata:
      annotations:
        workload.cattle.io/state: '{"b3ZoLWJtLTE=":"c-gw6hx:m-190d5b1abdb1","b3ZoLWJtLTM=":"c-gw6hx:m-93ddef52ec17","b3ZoLWRiLTE=":"c-mjbqh:m-cfa61f40f7d7"}'
      creationTimestamp: null
      labels:
        workload.user.cattle.io/workloadselector: deployment-cms-app
    spec:
      affinity: {}
      containers:
      - env:
        - <redacted>
        image: registry.ourdomain.com:5000/namespace/app:31848589
        imagePullPolicy: Always
        livenessProbe:
          failureThreshold: 3
          httpGet:
            path: /robots.txt
            port: 80
            scheme: HTTP
          initialDelaySeconds: 2
          periodSeconds: 5
          successThreshold: 1
          timeoutSeconds: 5
        name: app
        readinessProbe:
          failureThreshold: 3
          httpGet:
            path: /robots.txt
            port: 80
            scheme: HTTP
          initialDelaySeconds: 2
          periodSeconds: 5
          successThreshold: 2
          timeoutSeconds: 5
        resources: {}
        securityContext:
          allowPrivilegeEscalation: false
          privileged: false
          readOnlyRootFilesystem: false
          runAsNonRoot: false
        stdin: true
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        tty: true
        volumeMounts:
        - <redacted>
      dnsPolicy: ClusterFirst
      imagePullSecrets:
      - name: registry-secret
      nodeName: ovh-app-1
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
      volumes:
      - <redacted>
status:
  availableReplicas: 3
  conditions:
  - lastTransitionTime: 2018-10-02T05:18:54Z
    lastUpdateTime: 2018-10-02T05:18:54Z
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  - lastTransitionTime: 2018-10-03T03:48:46Z
    lastUpdateTime: 2018-10-03T04:48:25Z
    message: ReplicaSet "app-5bf7dbc69f" has successfully progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  observedGeneration: 45
  readyReplicas: 3
  replicas: 3
  updatedReplicas: 3

```

I can see the Deployments, ReplicaSets and Services as expected.

```
$ kubectl get deployment -n cms
NAME    DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
app     3         3         3            3           18h
redis   1         1         1            1           19h
```

```
$ kubectl get replicaset -n cms
NAME              DESIRED   CURRENT   READY   AGE
app-5bf7dbc69f    3         3         3       9h
app-7dc677d665    0         0         0       17h
app-849cc7c58d    0         0         0       18h
app-dd6cf6698     0         0         0       17h
redis-66985bf6c   1         1         1       19h
```

```
$ kubectl get service -n cms 
NAME                                       TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
app                                        ClusterIP   None            <none>        42/TCP    20h
ingress-e213c2b4c622329de7aa2c0c28dc37e5   ClusterIP   10.43.216.158   <none>        80/TCP    16s
redis                                      ClusterIP   None            <none>        42/TCP    21h

$ kubectl get service -n cms ingress-e213c2b4c622329de7aa2c0c28dc37e5 -o yaml
apiVersion: v1
kind: Service
metadata:
  annotations:
    field.cattle.io/targetWorkloadIds: '["deployment:cms:app"]'
  creationTimestamp: 2018-10-09T08:39:00Z
  labels:
    cattle.io/creator: norman
  name: ingress-e213c2b4c622329de7aa2c0c28dc37e5
  namespace: cms
  ownerReferences:
  - apiVersion: v1beta1/extensions
    controller: true
    kind: Ingress
    name: cms
    uid: 6617bfe4-c63f-11e8-b01c-9e111c023110
  resourceVersion: "1479746"
  selfLink: /api/v1/namespaces/cms/services/ingress-e213c2b4c622329de7aa2c0c28dc37e5
  uid: c4d9594b-cb9e-11e8-a6e1-9e111c023110
spec:
  clusterIP: 10.43.216.158
  ports:
  - port: 80
    protocol: TCP
    targetPort: 80
  selector:
    workloadID_ingress-e213c2b4c622329de7aa2c0c28dc37e5: "true"
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}

```

```
$ kubectl get ingress -n cms cms -o yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    field.cattle.io/creatorId: user-5jgmc
    field.cattle.io/ingressState: '{"Y21zL2Ntcy9kZXYuY21zLmFnZW50ZGVzaWduLmNvLnVrLy8vODA=":"deployment:cms:app","Y21zLWRldi1sZQ==":"cms:cms-dev-le"}'
    field.cattle.io/publicEndpoints: '[{"addresses":["94.237.50.126"],"port":443,"protocol":"HTTPS","serviceName":"cms:ingress-e213c2b4c622329de7aa2c0c28dc37e5","ingressName":"cms:cms","hostname":"dev.cms.agentdesign.co.uk","path":"/","allNodes":true}]'
  creationTimestamp: 2018-10-02T12:33:43Z
  generation: 15
  name: cms
  namespace: cms
  resourceVersion: "1479727"
  selfLink: /apis/extensions/v1beta1/namespaces/cms/ingresses/cms
  uid: 6617bfe4-c63f-11e8-b01c-9e111c023110
spec:
  rules:
  - host: dev.cms.agentdesign.co.uk
    http:
      paths:
      - backend:
          serviceName: ingress-e213c2b4c622329de7aa2c0c28dc37e5
          servicePort: 80
        path: /
  tls:
  - hosts:
    - dev.cms.agentdesign.co.uk
    secretName: cms-dev-le
status:
  loadBalancer:
    ingress:
    - ip: 94.237.50.126
    - ip: 94.237.51.162
    - ip: 94.237.54.20
    - ip: 94.237.54.24
    - ip: 94.237.54.26
```

```
$ kubectl get endpoints -n cms ingress-e213c2b4c622329de7aa2c0c28dc37e5
NAME                                       ENDPOINTS                                      AGE
ingress-e213c2b4c622329de7aa2c0c28dc37e5   10.42.0.157:80,10.42.3.146:80,10.42.4.168:80   10m
```

The Deployment scales down the 'old' ReplicaSet and scales up the 'new' ReplicaSet. I can see this is happening as expected.

```
$ kubectl get replicaset -n cms
NAME              DESIRED   CURRENT   READY   AGE
app-5bf7dbc69f    2         2         2       9h
app-7dc677d665    2         2         1       17h
app-849cc7c58d    0         0         0       18h
app-dd6cf6698     0         0         0       17h
redis-66985bf6c   1         1         1       19h
```

After a few seconds it's fully scaled...

```
$ kubectl get replicaset -n cms
NAME              DESIRED   CURRENT   READY   AGE
app-5bf7dbc69f    0         0         0       9h
app-7dc677d665    3         3         3       17h
app-849cc7c58d    0         0         0       18h
app-dd6cf6698     0         0         0       17h
redis-66985bf6c   1         1         1       19h
```

However, at this point the site is showing the 503 Service Temporarily Unavailable (nginx/1.13.12), because the nginx-ingress logs show that the ingress has been taken out of the generated nginx configurations...

```
W1009 07:01:57.954692       5 controller.go:769] Service "cms/ingress-e213c2b4c622329de7aa2c0c28dc37e5" does not have any active Endpoint.
I1009 07:01:57.955086       5 controller.go:169] Configuration changes detected, backend reload required.
I1009 07:01:57.955154       5 util.go:68] rlimit.max=1048576
I1009 07:01:57.955184       5 nginx.go:519] Maximum number of open file descriptors: 1047552
I1009 07:01:58.051641       5 nginx.go:626] NGINX configuration diff:
--- /etc/nginx/nginx.conf       2018-10-09 07:01:48.210045444 +0000
+++ /tmp/new-nginx-cfg780268064 2018-10-09 07:01:58.046093899 +0000
@@ -211,21 +211,12 @@
                
                keepalive 32;
                
-               server 10.42.3.138:80 max_fails=0 fail_timeout=0;
                server 10.42.0.148:80 max_fails=0 fail_timeout=0;
+               server 10.42.3.138:80 max_fails=0 fail_timeout=0;
                server 10.42.4.157:80 max_fails=0 fail_timeout=0;
                
        }
        
-       upstream cms-ingress-e213c2b4c622329de7aa2c0c28dc37e5-80 {
-               least_conn;
-               
-               keepalive 32;
-               
-               server 10.42.4.167:80 max_fails=0 fail_timeout=0;
-               
-       }
-       
        upstream db-ingress-231cd0bcc1b631a6142a515c3a0858e8-80 {
                least_conn;
                
@@ -657,7 +648,7 @@
                        
                        port_in_redirect off;
                        
-                       set $proxy_upstream_name "cms-ingress-e213c2b4c622329de7aa2c0c28dc37e5-80";
+                       set $proxy_upstream_name "";
                        
                        # enforce ssl on server side
                        if ($redirect_to_https) {
@@ -717,9 +708,8 @@
                        proxy_next_upstream                     error timeout;
                        proxy_next_upstream_tries               3;
                        
-                       proxy_pass http://cms-ingress-e213c2b4c622329de7aa2c0c28dc37e5-80;
-                       
-                       proxy_redirect                          off;
+                       # No endpoints available for the request
+                       return 503;
                        
                }
                
I1009 07:01:58.117524       5 controller.go:179] Backend successfully reloaded.
```

The reason for the 'Service ... does not have any active Endpoint' is as, according to [the docs](https://kubernetes.io/docs/tasks/debug-application-cluster/debug-service/#does-the-service-have-any-endpoints), because the 'endpoints controller has [not] found the correct Pods for your Service'.

```
$ kubectl get endpoints -n cms ingress-e213c2b4c622329de7aa2c0c28dc37e5
NAME                                       ENDPOINTS   AGE
ingress-e213c2b4c622329de7aa2c0c28dc37e5   <none>      12m
```

The advice given assumes the spec.selector field of the Service is not matching the metadata.labels values on your Pods. The `spec.selector` is:

```
spec:
  selector:
    workloadID_ingress-e213c2b4c622329de7aa2c0c28dc37e5: "true"
```

The metadata.labels on the Pods are:

```
metadata:
  labels:
    pod-template-hash: "1693867259"
    workload.user.cattle.io/workloadselector: deployment-cms-app
```

So, the docs are correct. But I'm still not sure why the label is not being set correctly on the workload.

**What you expected to happen**:

The endpoint controller to find active endpoints.

**How to reproduce it (as minimally and precisely as possible)**:

The same issue occurs on both our production and development clusters and was still present after we rebuilt both as fresh clusters and migrated things across. I suspect it's reproducible elsewhere.

**Anything else we need to know?**:

Not that I can think of currently.

**Environment**:
- Kubernetes version (use `kubectl version`): 

```
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.1", GitCommit:"4ed3216f3ec431b140b1d899130a69fc671678f4", GitTreeState:"clean", BuildDate:"2018-10-05T16:46:06Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.3", GitCommit:"a4529464e4629c21224b3d52edfe0ea91b072862", GitTreeState:"clean", BuildDate:"2018-09-09T17:53:03Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
```
Production - 3x Bare metal hosts with 64GB
Development - 5x Standard VMs with 8GB
```
- OS (e.g. from /etc/os-release):
```
VERSION="18.04.1 LTS (Bionic Beaver)"
```

- Kernel (e.g. `uname -a`):
```
Linux <hostnameredacted> 4.15.0-36-generic #39-Ubuntu SMP Mon Sep 24 16:19:09 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

```
- Install tools:

Clusters were deployed using Rancher2 (v2.0.8).

- Others:

N/A

**Results:**

Deployed service ends up with no active endpoints, causing nginx-ingress

**Comment by rossigee** [comment]
Hi @loganhz,

Thanks for looking into it.

Yes, this happens whenever we change the image version, whether in the UI, via `kubectl edit deployment` or via the following python code, which our CI system uses:

```
    def _upgrade_k8s_service(self, namespaceId, deploymentId, containerId, imageUri):
        # Use Kubernetes API call to update a deployment:
        # http://k8s-endpoint/apis/extensions/v1beta1/namespaces/<namespace>/deployments/<deployment>

        endpoint_url = "{0}/apis/apps/v1/namespaces/{1}/deployments/{2}".format(self.k8s_endpoint_url, namespaceId, deploymentId)
        headers = {
            'Content-Type': 'application/strategic-merge-patch+json',
            'Authorization': "Bearer " + self.k8s_token
        }
        payload = {
            "spec": {
                "template": {
                    "spec": {
                        "containers": [{
                            "name": containerId,
                            "image": imageUri
                        }]
                    }
                }
            }
        }

        # Upgrade the service with payload
        _logger.info("Updating container '{}' to use image '{}'... ".format(containerId, imageUri))
        r = requests.patch(endpoint_url, data=json.dumps(payload), headers=headers)
        res = r.json()
        if r.status_code != 200:
            raise Exception("Unexpected HTTP status code: {}".format(r.status_code))
```

The strange thing here is that it only seems to be reproducible when working with certain container images (basically supervisord+nginx+php-fpm running WordPress). We have various other deployments which use other images, which all work as expected.

> Not sure why workloadID_ingress-e213c2b4c622329de7aa2c0c28dc37e5: "true" is just missing in the pod.

This is indeed the question. I can't see any way that the contents of the container could (or should) be able to affect the setting of this label. However, I'm still not sure how/where that label gets set to see where/how it might fail to do so. I've been trying to figure it out from the source, but golang is not my strong point.

**Comment by rossigee** [comment]
@orangedeng - thanks for looking into this.

> The label workloadID_ingress-e213c2b4c622329de7aa2c0c28dc37e5 of pod is created by rancher, not from deployment or RS.

Could you point me to where in the code it is supposed to do this? It might help me better understand the sequence of events that could cause it to not be set as.

I've also noticed that while the label is not immediately set on deployment, if I go away for an hour or so, it will *eventually* set itself and start working again. What could cause an hour or more in delay?

> So if you have a way which can stably reproduce this problem, please let us know.

Unfortunately, the image in question contains proprietary/licensed code, so I can't readily make a reproducible test case.

**Comment by orangedeng** [comment]
@rossigee Sorry for the delay. The code in https://github.com/rancher/rancher/blob/master/pkg/controllers/user/targetworkloadservice/target_workload_service.go#L69. This controller will watch the service with annotation `field.cattle.io/targetWorkloadIds` and set labels to those pod related to the workloads.

**Comment by rossigee** [comment]
Thanks for that, @orangedeng.

This issue ended up being a significant blocker to our release plans, and even started happening randomly for no apparent reason (i.e. not just when we changed the image tag to update a deployment). For this reason, and because we were unable to make any progress in debugging it, we had to abandon Rancher2 and are now using a non-containerised stack for that particular application. We have now shut down the two clusters in which we could reproduce the issue, so I'll close this ticket for now. We may still revisit our containerisation plans later, and I'll re-open this if the issue still persists then.

**Comment by loganhz** [comment]
Let’s keep this issue open

**Comment by wanshaoyuan** [comment]
I did not find this problem in rancher v2.08 Kubernetes 1.11.2, rancher v2.1.1 Kubernetes 1.11.3, Kubernetes1.12.0, but I also heard from others that the choice of workload when creating ingress will have this problem, but service Is normal

**Comment by such** [comment]
@wanshaoyuan that's exactly what I've just experimented. Using workload in the ingress lead to a 503 after updating the image name in the GUI. Switching to service solved the issue.

**Comment by otreda** [comment]
+1 (Rancher 2.1.5 \ Kubernetes 1.11.6) same problem. Random effect after deploy

W0127 04:21:37.322027 7 controller.go:769] Service "default/ingress-a8aa9126f7bcd280b2704ab24c47ed76" does not have any active Endpoint.
I0127 04:21:37.331479 7 controller.go:169] Configuration changes detected, backend reload required.
I0127 04:21:38.546512 7 controller.go:179] Backend successfully reloaded.
I0127 04:21:40.637699 7 controller.go:169] Configuration changes detected, backend reload required.
I0127 04:21:41.800963 7 controller.go:179] Backend successfully reloaded.

**Comment by loganhz** [comment]
Thanks for the report. We still didn't find the way for how to reproduce it. We will look at it again.

**Comment by loganhz** [comment]
Hi @otreda 

Can you provide the yaml file for the service of the ingress, please?
![image](https://user-images.githubusercontent.com/21168270/51846205-14850e00-2354-11e9-8516-938ca4ad1eb0.png)

**Comment by loganhz** [comment]
And also the pod yaml of the target workloads please

![image](https://user-images.githubusercontent.com/21168270/51846243-2cf52880-2354-11e9-854b-810a620cad7d.png)

**Comment by otreda** [comment]
> Hi @otreda
> 
> Can you provide the yaml file for the service of the ingress, please?
> ![image](https://user-images.githubusercontent.com/21168270/51846205-14850e00-2354-11e9-8516-938ca4ad1eb0.png)

Hey

`apiVersion: v1
kind: Service
metadata:
  annotations:
    field.cattle.io/targetWorkloadIds: '["deployment:default:showcase"]'
  creationTimestamp: 2019-01-28T14:27:48Z
  labels:
    cattle.io/creator: norman
  name: ingress-1420a3ed5d8bef602c4f518d795cc1fe
  namespace: default
  ownerReferences:
  - apiVersion: v1beta1/extensions
    controller: true
    kind: Ingress
    name: dveri-lb4
    uid: e2abdb77-2308-11e9-9aa1-00155d08fd14
  resourceVersion: "7755873"
  selfLink: /api/v1/namespaces/default/services/ingress-1420a3ed5d8bef602c4f518d795cc1fe
  uid: e2d9ce56-2308-11e9-9aa1-00155d08fd14
spec:
  clusterIP: 10.43.221.185
  ports:
  - port: 8080
    protocol: TCP
    targetPort: 8080
  selector:
    workloadID_ingress-1420a3ed5d8bef602c4f518d795cc1fe: "true"
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}`

**Comment by otreda** [comment]
> And also the pod yaml of the target workloads please
> 
> ![image](https://user-images.githubusercontent.com/21168270/51846243-2cf52880-2354-11e9-854b-810a620cad7d.png)

`apiVersion: apps/v1beta2
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "38"
    field.cattle.io/creatorId: u-54l7t
    field.cattle.io/publicEndpoints: '[{"addresses":["195.211.101.190"],"port":443,"protocol":"HTTPS","serviceName":"default:ingress-49c6700ff4a49beb0c0ea7d0c8c30912","ingressName":"default:dveri-lb4","hostname":"test.dveri.ru","path":"/","allNodes":true},{"addresses":["195.211.101.190"],"port":443,"protocol":"HTTPS","serviceName":"default:ingress-2190f13a9fd595c40e8d7366ae05bc6b","ingressName":"default:dveri-lb4","hostname":"*.test.dveri.ru","path":"/","allNodes":true},{"addresses":["195.211.101.190"],"port":443,"protocol":"HTTPS","serviceName":"default:ingress-1420a3ed5d8bef602c4f518d795cc1fe","ingressName":"default:dveri-lb4","hostname":"review.dveri.ru","path":"/","allNodes":true}]'
  creationTimestamp: 2018-11-21T13:58:11Z
  generation: 67
  labels:
    cattle.io/creator: norman
    workload.user.cattle.io/workloadselector: deployment-default-showcase
  name: showcase
  namespace: default
  resourceVersion: "7755914"
  selfLink: /apis/apps/v1beta2/namespaces/default/deployments/showcase
  uid: 7bb6782b-ed95-11e8-8f68-00155d010235
spec:
  progressDeadlineSeconds: 600
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      workload.user.cattle.io/workloadselector: deployment-default-showcase
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 0
    type: RollingUpdate
  template:
    metadata:
      annotations:
        cattle.io/timestamp: 2018-12-17T13:44:08Z
      creationTimestamp: null
      labels:
        workload.user.cattle.io/workloadselector: deployment-default-showcase
    spec:
      containers:
      - env:
        - name: FORCE_RESTART_AT
          value: 2019-01-28T14:27:05UTC
        - name: MARKETING_NAME
          value: test.dveri.ru
        - name: NODE_ENV
          value: stage
        - name: WARMUP_CACHE
          value: "true"
        image: registry.dveri.ru:5000/dveri/showcase:release-1.0.3
        imagePullPolicy: Always
        name: showcase
        resources: {}
        securityContext:
          allowPrivilegeEscalation: false
          capabilities: {}
          privileged: false
          readOnlyRootFilesystem: false
          runAsNonRoot: false
        stdin: true
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        tty: true
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
status:
  availableReplicas: 1
  conditions:
  - lastTransitionTime: 2018-12-23T10:23:44Z
    lastUpdateTime: 2018-12-23T10:23:44Z
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  - lastTransitionTime: 2018-11-21T13:58:12Z
    lastUpdateTime: 2019-01-28T14:27:12Z
    message: ReplicaSet "showcase-5759498fb9" has successfully progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  observedGeneration: 67
  readyReplicas: 1
  replicas: 1
  updatedReplicas: 1`

**Comment by otreda** [comment]
> And also the pod yaml of the target workloads please
> 
> ![image](https://user-images.githubusercontent.com/21168270/51846243-2cf52880-2354-11e9-854b-810a620cad7d.png)

apiVersion: v1
kind: Pod
metadata:
  annotations:
    cattle.io/timestamp: 2018-12-17T13:44:08Z
    cni.projectcalico.org/podIP: 10.42.0.97/32
  creationTimestamp: 2019-01-28T14:27:08Z
  generateName: showcase-5759498fb9-
  labels:
    pod-template-hash: "1315054965"
    workload.user.cattle.io/workloadselector: deployment-default-showcase
    workloadID_ingress-49c6700ff4a49beb0c0ea7d0c8c30912: "true"
    workloadID_ingress-1420a3ed5d8bef602c4f518d795cc1fe: "true"
    workloadID_ingress-2190f13a9fd595c40e8d7366ae05bc6b: "true"
  name: showcase-5759498fb9-6bfwj
  namespace: default
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: ReplicaSet
    name: showcase-5759498fb9
    uid: caf589d3-2308-11e9-9aa1-00155d08fd14
  resourceVersion: "7987734"
  selfLink: /api/v1/namespaces/default/pods/showcase-5759498fb9-6bfwj
  uid: cb058af4-2308-11e9-9aa1-00155d08fd14
spec:
  containers:
  - env:
    - name: FORCE_RESTART_AT
      value: 2019-01-28T14:27:05UTC
    - name: MARKETING_NAME
      value: test.dveri.ru
    - name: NODE_ENV
      value: stage
    - name: WARMUP_CACHE
      value: "true"
    image: registry.dveri.ru:5000/dveri/showcase:release-1.0.3
    imagePullPolicy: Always
    name: showcase
    resources: {}
    securityContext:
      allowPrivilegeEscalation: false
      capabilities: {}
      privileged: false
      readOnlyRootFilesystem: false
      runAsNonRoot: false
    stdin: true
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    tty: true
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-7f6g2
      readOnly: true
  dnsPolicy: ClusterFirst
  nodeName: test
  priority: 0
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: default
  serviceAccountName: default
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - name: default-token-7f6g2
    secret:
      defaultMode: 420
      secretName: default-token-7f6g2
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: 2019-01-28T14:27:08Z
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: 2019-01-28T14:27:12Z
    status: "True"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: null
    status: "True"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: 2019-01-28T14:27:08Z
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://40fc1e4d147df408d63c48d219c13458aa588975b8c22301d7de8362441f7255
    image: registry.dveri.ru:5000/dveri/showcase:release-1.0.3
    imageID: docker-pullable://registry.dveri.ru:5000/dveri/showcase@sha256:abf70e89487f914c20a6d8eedf7f863ee629464768848ef75ea7dec91028a480
    lastState: {}
    name: showcase
    ready: true
    restartCount: 0
    state:
      running:
        startedAt: 2019-01-28T14:27:11Z
  hostIP: 195.211.101.190
  phase: Running
  podIP: 10.42.0.97
  qosClass: BestEffort
  startTime: 2019-01-28T14:27:08Z

**Comment by otreda** [comment]
and what about this problem?

**Comment by debianco** [comment]
Hey guys.
I've been kept from release and test an important product for my company and this is causing a big problem for my team. Could you please give me some advice about this issue?

Thanks

**Comment by loganhz** [comment]
@debianco 

Did you hit the `pods missing workloadID label - no active endpoints` issue?

**Comment by TheAifam5** [comment]
Hey! I'm getting the same issue, but only when sending POST requests. Appears again if I wait long enough. 

Logs below:
```
192.168.2.155 - [192.168.2.155] - - [23/Oct/2019:13:08:37 +0000] "GET /k8s/clusters/local/api/v1/namespaces/applications/pods/app-7d589948cb-g26zb/log?container=undefined&tailLines=500&follow=true&timestamps=true&previous=false HTTP/1.1" 400 221 "-" "REDACTED" 814 0.021 [cattle-system-rancher-80] 10.42.1.191:80 221 0.020 400 53198b5d6b2c6ae8ee8a7a0986a5cf30
192.168.2.155 - [192.168.2.155] - - [23/Oct/2019:13:09:47 +0000] "GET /v3/connect/config HTTP/2.0" 404 0 "-" "Go-http-client/2.0" 2793 0.003 [cattle-system-rancher-80] 10.42.2.242:80 0 0.000 404 f7d1b07e838c58ab5fe66438522eea62
192.168.2.155 - [192.168.2.155] - - [23/Oct/2019:13:11:47 +0000] "GET /v3/connect/config HTTP/2.0" 404 0 "-" "Go-http-client/2.0" 2793 0.002 [cattle-system-rancher-80] 10.42.0.249:80 0 0.000 404 b54518daf092a3f9ea6ad7de337774a7
192.168.2.155 - [192.168.2.155] - - [23/Oct/2019:13:15:58 +0000] "GET /v3/clusters/local/subscribe?sockId=2 HTTP/1.1" 101 3996 "-" "REDACTED" 639 742.725 [cattle-system-rancher-80] 10.42.0.249:80 0 742.724 101 90106440b3c9d4918eb3cf096ee2621c
192.168.2.155 - [192.168.2.155] - - [23/Oct/2019:13:16:24 +0000] "GET /v3/connect/config HTTP/2.0" 404 0 "-" "Go-http-client/2.0" 2799 0.003 [cattle-system-rancher-80] 10.42.2.242:80 0 0.004 404 df31794f026c2f55e0e236ff5f0c7d38
192.168.2.155 - [192.168.2.155] - - [23/Oct/2019:13:18:23 +0000] "GET /v3/connect/config HTTP/2.0" 404 0 "-" "Go-http-client/2.0" 2798 0.003 [cattle-system-rancher-80] 10.42.1.191:80 0 0.004 404 46f2482a613b9b360daa4f0c54dc389f
192.168.2.155 - [192.168.2.155] - - [23/Oct/2019:13:18:38 +0000] "GET / HTTP/1.1" 308 171 "-" "REDACTED" 355 0.000 [applications-ingress-96c051d5823628c5d7bf5ce1d7b6ed27-3000] - - - - cc26c73b9ebce81c5d595b0fd5a88fd8
192.168.2.155 - [192.168.2.155] - - [23/Oct/2019:13:20:24 +0000] "GET /v3/connect/config HTTP/2.0" 404 0 "-" "Go-http-client/2.0" 2799 0.003 [cattle-system-rancher-80] 10.42.2.242:80 0 0.004 404 da1349547da6d5c3a5e3d361d8af747b
192.168.2.155 - [192.168.2.155] - - [23/Oct/2019:13:22:23 +0000] "GET /v3/connect/config HTTP/2.0" 404 0 "-" "Go-http-client/2.0" 2798 0.002 [cattle-system-rancher-80] 10.42.1.191:80 0 0.004 404 7402129d674cfaab93f040849c71caed
192.168.2.155 - [192.168.2.155] - - [23/Oct/2019:13:24:23 +0000] "GET /v3/connect/config HTTP/2.0" 404 0 "-" "Go-http-client/2.0" 2798 0.002 [cattle-system-rancher-80] 10.42.2.242:80 0 0.000 404 caef528b9dd9bc0b0128a66188396537
192.168.2.155 - [192.168.2.155] - - [23/Oct/2019:13:26:24 +0000] "GET /v3/connect/config HTTP/2.0" 404 0 "-" "Go-http-client/2.0" 2799 0.004 [cattle-system-rancher-80] 10.42.0.249:80 0 0.004 404 f165c69e0007134b029b45711ae8ea8c
192.168.2.155 - [192.168.2.155] - - [23/Oct/2019:13:28:24 +0000] "GET /v3/connect/config HTTP/2.0" 404 0 "-" "Go-http-client/2.0" 2799 0.002 [cattle-system-rancher-80] 10.42.0.249:80 0 0.000 404 72af7ae6c257f8a8aa6919a1812379b4
W1023 13:29:11.660944       7 controller.go:826] Service "applications/ingress-96c051d5823628c5d7bf5ce1d7b6ed27" does not have any active Endpoint.
[23/Oct/2019:13:29:11 +0000]TCP200000.000
W1023 13:29:14.994526       7 controller.go:826] Service "applications/ingress-96c051d5823628c5d7bf5ce1d7b6ed27" does not have any active Endpoint.
```

**Comment by KeynesLee** [comment]
We are using the latest updates
rke 0.3.2 ( with default version of K8S - 1.15.5 ) , Rancher : 2.3.2 

We are still getting this issue.

**Comment by KeynesLee** [comment]
Just wonder ....

Is this issue fixed by Rancher v2.4.2 ?

**Comment by krumware** [comment]
This appears to have reared its ugly head on one of our v2.4.5 clusters
